docs/arch/adr-003-external-addons.md

```diff
@@ -0,0 +1,58 @@
+# ADR 3: kURL External Add-On Capability
+
+Today, a new add-on version cannot be released independently of the entire kURL project.
+As a result, when the github.com/replicatedhq/kots project is released, a kURL release must follow.
+Any changed staged for a kURL release must be ready or the KOTS release will be blocked.
+Additionally, a maintainer of the kURL project must be available to make the release.
+
+## Decision
+
+We will add support for releasing add-ons independent of the kURL project.
+Some add-ons will continue to be released with kURL.
+The KOTS add-on will be the first to leverage this new feature.
+
+An external add-on will publish a versions.json file to a publicly accessible URL with metadata describing versions it has published.
+This data will contain the version, url of the add-on package, and the kURL version compatibility range.
+
+versions.json
+
+```json
+[
+  {
+    "version": "1.85.0",
+    "url": "https://kots-kurl-addons-production-1658439274.s3.amazonaws.com/kotsadm-1.85.0-a482541.tar.gz",
+    "kurlVersionCompatibilityRange": ">= v2022.10.05-0"
+  }
+]
+```
+
+kURL will make available a set of tools (GitHub actions to start) to aid external add-ons in building and testing external add-on packages.
+
+kURL automation will periodically poll each external add-on's published versions.json file and copy the new add-on package versions to its object storage bucket.
+The add-on package will be stored external to any specific kURL version, as kURL versions are immutable.
+When retrieving add-on versions from the kURL API, the API will append all internal versions with external ones that satisfy the `kurlVersionCompatibilityRange`.
+
+When a kURL spec is resolved as part of the script rendering process, external add-on package versions will be rendered including the `s3Override` property with reference to the package.
+kURL will use this property to identify the source and download the add-on package when installing the add-on in the end user's environment.
+
+```yaml
+spec:
+  kotsadm:
+    version: 1.85.0
+    s3Override: https://s3.kurl.sh/external/kotsadm-1.85.0-a482541.tar.gz
+```
```

Comment:
Can we move these details to the **Consequences** section? The **Decision** subheading should be short and concise.

Reply:
Yes, there's no hard and fast rule for additional context in the decision section but in my experience and in practice, this section should only state the final decision. This makes the _Decision_ subheading cleaner and concise for the purpose of better readability when skimming through an ADR repository (and enables ADR tooling to just pull decision verbiage only). Moreover, the why&how details seem to fit better in the first subheading (general context) or perhaps in a new subheading which is all useful information which leads to a decision.

Reply:
i prefer to keep context as the "why". I feel like the current template there is no place for "how". I like the idea of a new subheading. i will look at other adr templates and make a proposal.

Reply:
@rrpolanco proposed changes included in this commit

https://github.com/replicatedhq/kURL/pull/3543/commits/1728e3d31dd5ef653899fe1cb9fec8cd19bc027c